src/pdf_parser.cpp

```diff
@@ -3638,14 +3638,29 @@ struct PDFParser::Implementation
 			{
 				m_root_ref = new PDFReferenceCall(*this);
 				m_info_ref = new PDFReferenceCall(*this);
-				readReferenceData();
-				if (m_got_info)
-					m_info = m_info_ref->getDictionary();
-				if (m_got_root)
-					m_root_dictionary = m_root_ref->getDictionary();
-				if (!m_root_dictionary)
-					throw doctotextex::CustomException("Root dictionary is missing!");
+				
+				try {
+					readReferenceData();
+					if (m_got_info)
+						m_info = m_info_ref->getDictionary();
+					if (m_got_root)
+						m_root_dictionary = m_root_ref->getDictionary();
+					if (!m_root_dictionary)
+						throw doctotextex::CustomException("Root dictionary is missing!");
 				m_metadata = m_root_dictionary->getObjAsStream("Metadata");
+				} catch (Exception &e) {
```

Comment:
Did you mean `std::exception`? I don't see where `Exception` is defined (sorry if I'm missing something it's been awhile since I looked at this codebase)

Reply:
Sorry,It's my fault.@_@